ansible_rulebook/util.py

```diff
@@ -380,3 +380,46 @@ def validate_url(url: str, url_type: str) -> bool:
     if url_type == "controller":
         return res.scheme in ["http", "https"] and bool(res.netloc)
     return res.scheme in ["ws", "wss"] and bool(res.netloc)
+
+
+# If the following values exist in a key,
+# replace the value of the key with the MASKED_VARIABLE
+KEYS_TO_FILTER = [
+    "token",
+    "password",
+]
+MASKED_VARIABLE = "******"
+
+
+def mask_sensitive_variable(key: str, val: str):
+    """
+    Takes in a key and value, if they key contains a
+    substring in the KEYS_TO_FILTER, mask the value.
+    Otherwise return the value.
+    """
+    if any(
+        filtered_key.casefold() in str(key).casefold()
+        for filtered_key in KEYS_TO_FILTER
+    ):
+        return MASKED_VARIABLE
+    else:
+        return val
+
+
+def mask_sensitive_variable_values(variables):
```

Comment:
@zkayyali812 Can we add types in here for variables

Reply:
@mkanoor Updated, does this look better now?